index.d.ts

```diff
@@ -3,10 +3,9 @@
 
 export = update
 
-// declare function update<T>(data: T, query: update.Query<T>): T
 declare function update<T>(
   data: ReadonlyArray<T>,
-  query: ArrayOperators<T>,
+  query: ArrayOperators<T> | any,
```

Comment:
I think `any` in this case could be replaced with `{ [key: number]: Query<T> }`.
However, example from #84 used `{ $splice: [[1, 1, 13, 14]] }` which is currently not a valid query, since the types declare `{$splice: Array<[number, number]>}` which doesn't allow for 4 element arrays. Either the type is wrong, or the example should use `[[1, 1], [13, 14]]` instead.

Reply:
Additionally, I don't think this change is sufficient to fix all breaking changes.

For example, given the following interfaces, the update below is valid given v2.6.2, but not 2.6.3+, and the proposed change does not fix this case (presuming dataSource is an instance of IFDSource):

Query:  update(dataSource, {sources:{0:{definition:{0:{$set: newDefinition}}, foo: {$set: "junk"}}}});
```
export interface IFDSource {
    sources: Array<IFSource>;
}

export interface IFSource {
    foo: string;
    definition: Array<IFDSourceDefinition>;
}

export interface IFDSourceDefinition {
    x: string;
    source: string;
}
```

However, the solution proposed by @sohymg does fix this issue.